syft/frameworks/torch/mpc/falcon.py

```diff
@@ -0,0 +1,82 @@
+import torch
+
+from syft.frameworks.torch.tensors.interpreters.replicated_shared import ReplicatedSharingTensor
+from syft.workers.base import BaseWorker
+
+from typing import List
+
+
+class Falcon(object):
+    @staticmethod
+    def _share_random_tensor(
+        field: int, workers: List[BaseWorker]
+    ) -> (ReplicatedSharingTensor, ReplicatedSharingTensor):
+        """
+        TODO: Need to move this in Primitive Store:
+        Share a random tensor with the owners of some shares
+        """
+        c = torch.randint(2, size=(1,))
+        c_share_L = c.share(*workers, protocol="falcon", field=field)
+        c_share_2 = c.share(*workers, protocol="falcon", field=2)
+        return c_share_2, c_share_L
+
+    @staticmethod
+    def select_share(
+        b_share: ReplicatedSharingTensor,
+        x_share: ReplicatedSharingTensor,
+        y_share: ReplicatedSharingTensor,
+    ) -> ReplicatedSharingTensor:
+        """ Performs select share protocol
+        If the bit b_share is 0, x_sh is returned
+        If the bit b_share is 1, y_sh is returned
+
+        Args:
+            x_share (ReplicatedSharingTensor): the first share to select
+            y_share (ReplicatedSharingTensor): the second share to select
+            b_share (ReplicatedSharingTensor): the bit to take into consideration
+                when selecting between x or y
+
+        Return:
+            x_share or b_share depending on the value of b_share
+        """
+
+        field = x_share.field
+        players = x_share.players
+        shape = x_share.shape
+        c_share_2, c_share_L = Falcon._share_random_tensor(field, players)
+
+        diff = y_share - x_share
+        xor_b_c = b_share.xor(c_share_2)
+
+        ones_share = ReplicatedSharingTensor.one_shares(players, field)
+        if xor_b_c.reconstruct().item():
+            d_share = ones_share - c_share_L
+        else:
+            d_share = c_share_L
+
+        selected_share = diff * d_share + x_share
+        return selected_share
+
+    @staticmethod
+    def evaluate(
+        x_share: ReplicatedSharingTensor, beta_share: ReplicatedSharingTensor
+    ) -> ReplicatedSharingTensor:
+        """ Performs the computation (-1)^beta * x
+
+        Args:
+            x_share (ReplicatedSharingTensor): share to "evaluate"
+            beta_share (ReplicatedSharingTensor): the reconstructed value should be in {0, 1}
+
+        Return:
+            A RST that is (-1)^beta * x
+        """
+        field = x_share.field
+        players = x_share.players
+        shape = x_share.shape
+
+        pow_sh = (
+            ReplicatedSharingTensor.one_shares(players, field=field, shape=shape) - beta_share * 2
```

Comment:
I don't think that you need to share a 1 for that. a much more efficient way will be to do (see Falcon paper, Linear operations): 
party1 share =  1-2[beta1]
party2 share =  -2[beta2]
party3 share =  -2[beta3]

Reply:
Yep - I left a comment there - basically doing this I think becomes a "little" intrusive with the RST - but it is worth thinking if we can rewrite this - hopefully, we will finish falcon soon and we will start benchmarking it and see where we can "get some juice"